platforms/core-configuration/declarative-dsl-provider/src/testFixtures/groovy/org/gradle/internal/declarativedsl/settings/SoftwareTypeFixture.groovy

```diff
@@ -766,6 +788,12 @@ trait SoftwareTypeFixture {
                     @Restricted
                     public abstract Property<String> getId();
 
+                    @Restricted
+                    public abstract DirectoryProperty getDir();
+
+                    @Restricted
+                    public abstract RegularFileProperty getFile();
```

Comment:
🤔 As far as I can see, we don't support `Property<RegularFile>` nor `Property<Directory>`.
If we stick to that decision for now it would be good to have tests to know how we behave when software types have such properties and assess if this is acceptable.

Reply:
There is now an integration test showing that they work.